src/main/java/com/avispl/symphony/dal/infrastructure/management/samsung/smartthings/SamsungSmartThingsAggregatorCommunicator.java

```diff
@@ -0,0 +1,3177 @@
+		}
+
+		// populate device control base on active room
+		for (Device device : cachedDevicesAfterPollingInterval.values()) {
+
+			// filter device by value of ActiveRoom dropdown list
+			String roomId = getDefaultValueForNullData(device.getRoomId(), SmartThingsConstant.EMPTY);
+			if (roomId.equals(currentRoomId)) {
+				Optional<List<DetailViewPresentation>> actions = Optional.ofNullable(device.getPresentation())
+						.map(DevicePresentation::getDashboardPresentations)
+						.map(DashboardPresentation::getActions);
+
+				// populate control
+				if (actions.isPresent() && !actions.get().isEmpty()) {
+					DetailViewPresentation action = actions.get().get(0);
+					switch (action.getDisplayType()) {
+						case DeviceDisplayTypesMetric.STAND_BY_POWER_SWITCH:
+						case DeviceDisplayTypesMetric.TOGGLE_SWITCH:
+						case DeviceDisplayTypesMetric.SWITCH:
+							String onLabel = getDefaultValueForNullData(action.getStandbyPowerSwitch().getCommand().getOn(), SmartThingsConstant.ON);
+							String offLabel = getDefaultValueForNullData(action.getStandbyPowerSwitch().getCommand().getOff(), SmartThingsConstant.OFF);
+							String currentVale = getDefaultValueForNullData(action.getStandbyPowerSwitch().getValue(), SmartThingsConstant.OFF);
+
+							addAdvanceControlProperties(advancedControllableProperties,
+									createSwitch(stats, AggregatorGroupControllingMetric.DEVICES_DASHBOARD.getName() + device.getName(), currentVale, offLabel, onLabel));
+							unusedDeviceControlKeys.add(AggregatorGroupControllingMetric.DEVICES_DASHBOARD.getName() + device.getName());
+							isEmptyDevicesControl = false;
+							break;
+						case DeviceDisplayTypesMetric.PUSH_BUTTON:
+							addAdvanceControlProperties(advancedControllableProperties,
+									createButton(stats, AggregatorGroupControllingMetric.DEVICES_DASHBOARD.getName() + device.getName(), SmartThingsConstant.PUSH, SmartThingsConstant.PUSHING));
+							unusedDeviceControlKeys.add(AggregatorGroupControllingMetric.DEVICES_DASHBOARD.getName() + device.getName());
+
+							isEmptyDevicesControl = false;
+							break;
+						default:
+							if (logger.isWarnEnabled()) {
+								logger.warn(String.format("Unexpected device display type: %s", action.getDisplayType()));
+							}
+							break;
+					}
+				}
+			}
+		}
+		// populate message when room have no device after filtering
+		if (isEmptyDevicesControl) {
+			stats.put(AggregatorGroupControllingMetric.DEVICES_DASHBOARD.getName() + AggregatorGroupControllingMetric.MESSAGE.getName(), "There are no devices available in this room or location");
+		} else {
+			stats.remove(AggregatorGroupControllingMetric.DEVICES_DASHBOARD.getName() + AggregatorGroupControllingMetric.MESSAGE.getName());
+		}
+	}
+
+	/**
+	 * This method is used to validate if the locations contain the No room Assigned
+	 *
+	 */
```

Comment:
missing @return tag. Also, is there a reason this method is public?

Reply:
I fixed it